crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -331,6 +331,28 @@ impl AuthorityEpochTables {
             .map(|(_k, v)| v)
             .collect()
     }
+
+    pub fn checkpoint_epoch_start_configuration(
```

Comment:
I don't actually understand why this table needs to be checkpointed.
cc @andll

Reply:
That seems a bug to me?

Reply:
It seems like it needs `epoch_start_configuration` from somewhere in the `AuthorityPerEpochStore::new` function. Since it is not getting passed (epoch > 0), it expects to find it in the table (which is written to during reconfig). The logic seems reasonable to me but maybe I am missing something. What is the exact bug here and what should we do to fix?
```
 // (2) For other epochs, this is updated when AuthorityPerEpochStore
        // is initialized during epoch change
```

Reply:
I think we should make it able to construct epoch_start_configuration based on the current store state, instead of relying on a previous epoch store.

Reply:
The point at which db snapshots are taken during reconfiguration, the system state object is already updated in the db. When we restore, we restore it to the same point (just before reconfiguration) and older system state object is updated. 
Keeping epoch_start_configuration adds little overhead to the checkpoint process (its just one key literally) and allows restoring by a trivial directory copy (constructing epoch_start_configuration from other db state might require a more complicated restore process which is why I like the current approach which keep things simple). But I am happy to do the right thing (but may be we can do it in a different PR?)